### PR TITLE
[INLONG-5114][Manager] Add role permission verification function

### LINKED ISSUE #5114: [Improve][Manager] Add role permission verification function
### Description

Add role permission verification function

</body></html>
<table>
	<tr>
		<th>账号角色授权</th>
		<th>operation</th>
		<th>管理员 </th>
                 <th>普通用户  </th>
              <th>其他约束  </th>
	<tr/>
	<tr>
		<td  rowspan="7">用户管理 </td>
		<td> 查看用户</td>
		<td>✅</td>
               <td>仅查看自己 </td>
	</tr>
		<tr>
		<td>新增用户</td>
		<td>✅</td>
		<td></td>
                 <td> name冲突检查等<td>
	</tr>
		<tr>
		<td>修改用户名</td>
		<td>✅ </td>
		<td></td>
		<td>name冲突检查等</td>
	</tr>
	</tr>
		<tr>
		<td>修改用户密码</td>
		<td>✅ </td>
		<td>仅修改自己</td>
		<td>需要验证老密码</td>
	</tr>
	</tr>
		<tr>
		<td>修改用户角色</td>
		<td>✅ </td>
		<td></td>
		<td>管理员不能将自己被改为普通账号, 
能修改其他管理员账号为普通账号</td>
	</tr>
		<tr>
		<td>修改账户有效期</td>
		<td>✅ </td>
		<td>只允许减少有效期</td>
		<td></td>
	</tr>
		<tr>
		<td>删除用户</td>
		<td>✅ </td>
		<td></td>
		<td>不能删自己，能删其他管理员账号</td>
	</tr>
	</tr>
		<tr>
                 <td  rowspan="2">审批人员管理 </td>
		<td>查看审批责任人</td>
		<td>✅ </td>
		<td></td>
		<td></td>
	</tr>
	</tr>
		<tr>
		<td>修改审批责任人</td>
		<td>✅ </td>
		<td></td>
		<td></td>
	</tr>
</tr>
		<tr>
                 <td  rowspan="3">审批管理 </td>
		<td>查看待审批</td>
		<td>✅ </td>
		<td>全部申请单</td>
		<td>自己的申请单</td>
	</tr>
	</tr>
		<tr>
		<td>消费组审批</td>
		<td>✅ </td>
		<td>自己需要审批的</td>
		<td></td>
	</tr>
	</tr>
		<tr>
		<td>group审批</td>
		<td>✅ </td>
		<td>自己需要审批的</td>
		<td></td>
	</tr>
	</tr>
		<tr>
                 <td  rowspan="4">集群管理 </td>
		<td>pulsar删改查</td>
		<td>✅ </td>
		<td>仅可操作自己负责的</td>
		<td rowspan="4">name冲突检查等
删除标签时判断当前标签是否被group所使用，是则不能删除，否则删除各个集群中的该标签后对该标签进行删除</td>
	</tr>
	</tr>
		<tr>
		<td>dataproxy删改查</td>
		<td>✅ </td>
		<td>自己需要审批的</td>
	</tr>
	</tr>
		<tr>
		<td>tube删改查</td>
		<td>✅ </td>
		<td>自己需要审批的</td>
	</tr>
	</tr>
		<tr>
		<td>标签管理删改查</td>
		<td>✅ </td>
		<td>自己需要审批的</td>
		<td></td>
	</tr>

</table>
</body></html>
<table>
	<tr>
		<th>Account role authorization</th>
		<th>Operation</th>
		<th>Admin </th>
                 <th>Ordinary users </th>
              <th>Other constraints  </th>
	<tr/>
	<tr>
		<td  rowspan="7">user management </td>
		<td>get user info</td>
		<td>✅</td>
               <td>only get his own </td>
	</tr>
		<tr>
		<td>add user info</td>
		<td>✅</td>
		<td></td>
                 <td> <td>
	</tr>
		<tr>
		<td>update user info</td>
		<td>✅ </td>
		<td></td>
		<td>name conflict check, etc</td>
	</tr>
	</tr>
		<tr>
		<td>update user info password</td>
		<td>✅ </td>
		<td>only update his own</td>
		<td>need to verify old password</td>
	</tr>
	</tr>
		<tr>
		<td>update user role</td>
		<td>✅ </td>
		<td></td>
		<td>administrators cannot change themselves to ordinary accounts,
can modify other administrator accounts to ordinary accounts</td>
	</tr>
		<tr>
		<td>update user valid days</td>
		<td>✅ </td>
		<td>only allow to reduce the valid days</td>
		<td></td>
	</tr>
		<tr>
		<td>delete user </td>
		<td>✅ </td>
		<td></td>
		<td>you cannot delete yourself. you can delete other administrator accounts</td>
	</tr>
	</tr>
		<tr>
                 <td  rowspan="2">approvers management </td>
		<td>get approval responsible person</td>
		<td>✅ </td>
		<td></td>
		<td></td>
	</tr>
	</tr>
		<tr>
		<td>update approval responsible person</td>
		<td>✅ </td>
		<td></td>
		<td></td>
	</tr>
</tr>
		<tr>
                 <td  rowspan="3">approval management </td>
		<td>get pending approval</td>
		<td>✅ </td>
		<td></td>
		<td>only get his own</td>
	</tr>
	</tr>
		<tr>
		<td>consumption approval</td>
		<td>✅ </td>
		<td>the responsible person is himself</td>
		<td></td>
	</tr>
	</tr>
		<tr>
		<td>group approval</td>
		<td>✅ </td>
		<td>the responsible person is himself</td>
		<td></td>
	</tr>
	</tr>
		<tr>
                 <td  rowspan="4">cluster management </td>
		<td>delete、update、get pulsar cluster info</td>
		<td>✅ </td>
		<td>the responsible person is himself</td>
		<td rowspan="4">name conflict check, etc
when deleting a tag, judge whether the current tag is used by the group. If yes, it cannot be deleted. Otherwise, delete the tag after deleting it in each cluster</td>
	</tr>
	</tr>
	</tr>
		<tr>
		<td>delete、update、get dataproxy cluster info</td>
		<td>✅ </td>
		<td>the responsible person is himself</td>
	</tr>
	</tr>
		<tr>
		<td>delete、update、get tube cluster info</td>
		<td>✅ </td>
		<td>the responsible person is himself</td>
	</tr>
	</tr>
		<tr>
		<td>delete、update、get cluster tag info</td>
		<td>✅ </td>
		<td>the responsible person is himself</td>
	</tr>

</table>
InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
### Prepare a Pull Request

- Fixes #5114 

### Motivation

Add role permission verification function

### Modifications
Add role permission verification function

### Verifying this change

- [X] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterService.java [hunk 1]
```diff
@@ -29,13 +29,22 @@
...
  */
 public interface InlongClusterService {
+ 
+    /**
+     * Get user info by user name
+     *
+     * @param username username
+     * @return user info
+     */
+    UserEntity getByName(String username);
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterService.java:47]: 1. `getByUsername` is better.
2. This method should move to `UserService`, which is more object-oriented.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java [hunk 2]
```diff
@@ -218,6 +229,9 @@ public Boolean deleteTag(Integer id, String operator) {
             LOGGER.error("inlong cluster tag not found by id={}", id);
             return false;
         }
+        String[] inChargesArr = exist.getInCharges().split(",");
+        Preconditions.checkTrue(isInCharge(inChargesArr, operator),
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java:233]: The `String[] inChargesArr = exist.getInCharges().split(",");` can be merged into `isInCharge()` method.

BTW, the `isInCharge()` method seems can refactor as a util method.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -48,6 +48,8 @@
 import java.util.Date;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java:52]: No need for more than one blank line.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java (hunk 2 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java:233]: The `exist` have  many types, like `InlongClusterEntity`、`InlongClusterNodeEntity`、`InlongClusterTagEntity`.

**Review by woofyzhao — APPROVED** [review]
LGTM

**Review by healchow — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Review by GanfengTan — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**